File: Problem006/Problem006.java
class Problem006 {
    public static void main(String[] args) {
        long sumOfSquares = 0L;
        long squareOfSums = 0L;
        for(long x = 0L; x <= 100L; x++) {
            sumOfSquares += Math.pow(x, 2);
            squareOfSums += x;
        }
        System.out.println(Math.pow(squareOfSums, 2) - sumOfSquares);
    }
}

File: Problem006/Problem006.hs
main = print(squareOfSums(100) -  sumOfSquares(100))

squareOfSums::Int->Int
squareOfSums n = ((n * (n + 1)) `div` 2) * ((n * (n + 1)) `div` 2)

sumOfSquares::Int->Int
sumOfSquares n = (n * (n + 1) * ((2 * n) + 1) `div` 6)
